### PR TITLE
MCR-2808 MCRURNUtils.getDNBRegisterDate now use URN REST-API V2

### PR DESCRIPTION
**Pull request by chiliego**
[Link to jira](https://mycore.atlassian.net/browse/MCR-2808).

### CODE REVIEW

**Review by sebhofmann — CHANGES_REQUESTED** [review]
Please make MCRDNBURNRestClient usable without a bundleProvider or make getRegistrationInfo static.

**Comment by chiliego** [comment]
@sebhofmann can you explain your comment in more detail, why "make MCRDNBURNRestClient usable without a bundleProvider or make getRegistrationInfo static"?

File: mycore-pi/src/main/java/org/mycore/pi/urn/rest/MCRDNBURNRestClient.java [hunk 1]
```diff
@@ -161,6 +164,36 @@ public Optional<Date> register(MCRPIRegistrationInfo urn) {
...
 
+    public Optional<JsonObject> getRegistrationInfo(MCRPIRegistrationInfo urn) {
+        String identifier = urn.getIdentifier();
+        return getRegistrationInfo(identifier);
+    }
+
+    public static Optional<JsonObject> getRegistrationInfo(String identifier) {
+        MCRDNBURNRestClient mcrurnClient = new MCRDNBURNRestClient(
+                u -> MCRURNJsonBundle.instance(u, MCRDerivateURNUtils.getURL(u)));
+        String url = mcrurnClient.getBaseServiceURL() + "/urn/" + identifier;
```
Comment (sebhofmann) [inline mycore-pi/src/main/java/org/mycore/pi/urn/rest/MCRDNBURNRestClient.java]: The method getBaseServiceURL() does not rely on any field of the class so you could make it static or add a static variant which is used by the non static version.

File: mycore-pi/src/main/java/org/mycore/pi/urn/rest/MCRDNBURNRestClient.java [hunk 2]
```diff
@@ -161,6 +164,36 @@ public Optional<Date> register(MCRPIRegistrationInfo urn) {
...
+                try {
+                    Reader reader = new InputStreamReader(entity.getContent());
+                    JsonElement jsonElement = JsonParser.parseReader(reader);
+                    return Optional.of(jsonElement.getAsJsonObject());
+                } catch (Exception e) {
+                    LOGGER.error("Could not read Response from " + url);
+                }
+                break;
+            default:
+                mcrurnClient.logFailure("", response, statusCode, identifier, url);
```
Comment (sebhofmann) [inline mycore-pi/src/main/java/org/mycore/pi/urn/rest/MCRDNBURNRestClient.java]: The method logFailure() does not rely on any field of the class so you could make it static or add a static variant which is used by the non static version.

File: mycore-pi/src/main/java/org/mycore/pi/urn/rest/MCRDNBURNRestClient.java [hunk 3]
```diff
@@ -161,6 +164,36 @@ public Optional<Date> register(MCRPIRegistrationInfo urn) {
         return Optional.empty();
     }
 
+    public Optional<JsonObject> getRegistrationInfo(MCRPIRegistrationInfo urn) {
+        String identifier = urn.getIdentifier();
+        return getRegistrationInfo(identifier);
+    }
+
+    public static Optional<JsonObject> getRegistrationInfo(String identifier) {
+        MCRDNBURNRestClient mcrurnClient = new MCRDNBURNRestClient(
```
Comment (sebhofmann) [inline mycore-pi/src/main/java/org/mycore/pi/urn/rest/MCRDNBURNRestClient.java]: If you make some other methods static you can remove the instance of the class(MCRDNBURNRestClient) and the weird constructor call is not needed.

**Review by sebhofmann — CHANGES_REQUESTED** [review]

**Review by sebhofmann — APPROVED** [review]
apart from the codacy errors the changes looks good now

**Comment by kkrebs** [comment]
please fix forbidden api error
Error:  Forbidden method invocation: java.io.InputStreamReader#<init>(java.io.InputStream) [Uses default charset]
Error:    in org.mycore.pi.urn.rest.MCRDNBURNRestClient (MCRDNBURNRestClient.java:181)
Error:  Scanned 107 class file(s) for forbidden API invocations (in 0.21s), 1 error(s).

... see build log

**Pull request merged**